pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -693,6 +746,80 @@ func addRoleConfig(config map[string]interface{}, controlPlane *rkev1.RKEControl
 		config["disable-etcd"] = true
 	}
 
+	// If this is a control-plane node, then we need to set arguments/(and for RKE2, volume mounts) to allow probes
+	// to run.
+	if isControlPlane(machine) {
+		renderedKubeControllerManagerCertDir := fmt.Sprintf(KubeControllerManagerCertDir, runtime)
+		rke2KCMCertDirMount := fmt.Sprintf("%s:%s", renderedKubeControllerManagerCertDir, renderedKubeControllerManagerCertDir)
+		kcmCertDirArg := fmt.Sprintf("%s=%s", CertDirArgument, renderedKubeControllerManagerCertDir)
+		kcmSecurePortArg := fmt.Sprintf("%s=%s", SecurePortArgument, KubeControllerManagerDefaultSecurePort)
+		if v, ok := config[KubeControllerManagerArg]; ok {
+			tlsCF := getArgValue(v, TLSCertFileArgument, "=")
+			if tlsCF == "" {
+				certDir := getArgValue(v, CertDirArgument, "=")
+				if certDir == "" {
+					config[KubeControllerManagerArg] = appendToInterface(config[KubeControllerManagerArg], kcmCertDirArg)
+					if runtime == rancherruntime.RuntimeRKE2 {
+						config["kube-controller-manager-extra-mount"] = appendToInterface(config["kube-controller-manager-extra-mount"], rke2KCMCertDirMount)
+					}
+				} else {
+					if runtime == rancherruntime.RuntimeRKE2 {
+						config["kube-controller-manager-extra-mount"] = appendToInterface(config["kube-controller-manager-extra-mount"], fmt.Sprintf("%s:%s", certDir, certDir))
+					}
+				}
+			} else {
+				if runtime == rancherruntime.RuntimeRKE2 {
+					dir := filepath.Dir(tlsCF)
+					config["kube-controller-manager-extra-mount"] = appendToInterface(config["kube-controller-manager-extra-mount"], fmt.Sprintf("%s:%s", dir, dir))
+				}
+			}
+			sPA := getArgValue(v, SecurePortArgument, "=")
+			if sPA == "" {
+				config[KubeControllerManagerArg] = appendToInterface(config[KubeControllerManagerArg], kcmSecurePortArg)
+			}
+		} else {
+			config[KubeControllerManagerArg] = []string{kcmCertDirArg, kcmSecurePortArg}
+			if runtime == rancherruntime.RuntimeRKE2 {
+				config["kube-controller-manager-extra-mount"] = appendToInterface(config["kube-controller-manager-extra-mount"], rke2KCMCertDirMount)
+			}
+		}
```

Comment:
This is a simplification: if you just keep track of what `rke2KCMCertDirMount` should be, then you can add it at the end, if necessary. Also, I think making `kube-controller-manager-extra-mount` a constant like the others is good.
```suggestion
	if isControlPlane(machine) {
		renderedKubeControllerManagerCertDir := fmt.Sprintf(KubeControllerManagerCertDir, runtime)
		rke2KCMCertDirMount := fmt.Sprintf("%s:%s", renderedKubeControllerManagerCertDir, renderedKubeControllerManagerCertDir)
		kcmCertDirArg := fmt.Sprintf("%s=%s", CertDirArgument, renderedKubeControllerManagerCertDir)
		kcmSecurePortArg := fmt.Sprintf("%s=%s", SecurePortArgument, KubeControllerManagerDefaultSecurePort)
		if v, ok := config[KubeControllerManagerArg]; ok {
			tlsCF := getArgValue(v, TLSCertFileArgument, "=")
			if tlsCF == "" {
				certDir := getArgValue(v, CertDirArgument, "=")
				if certDir == "" {
					config[KubeControllerManagerArg] = appendToInterface(config[KubeControllerManagerArg], kcmCertDirArg)
				} else {
					rke2KCMCertDirMount = fmt.Sprintf("%s:%s", certDir, certDir))
				}
			} else {
				 rke2KCMCertDirMount = filepath.Dir(tlsCF)
			}
			sPA := getArgValue(v, SecurePortArgument, "=")
			if sPA == "" {
				config[KubeControllerManagerArg] = appendToInterface(config[KubeControllerManagerArg], kcmSecurePortArg)
			}
		} else {
			config[KubeControllerManagerArg] = []string{kcmCertDirArg, kcmSecurePortArg}
		}
	 if runtime == rancherruntime.RuntimeRKE2 {
		  config["kube-controller-manager-extra-mount"] = appendToInterface(config["kube-controller-manager-extra-mount"], rke2KCMCertDirMount)
	 }
```

Reply:
Simplified